desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/OverflowingItem.tsx

```diff
@@ -0,0 +1,93 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useEffect, useRef, useState } from 'react';
+import { Button } from 'antd';
+
+import OverflowTooltip from './OverflowTooltip';
+import './OverflowingItem.scss';
+
+enum OverflowingComponentType {
+  breadcrumb,
+  menu
+}
+
+type OverflowingComponentTypeString = keyof typeof OverflowingComponentType;
+
+interface OverflowingItemProps {
+  label: string;
+  url: string;
+  handleFilePathChange: (path: string) => void;
+  componentType: OverflowingComponentTypeString;
+}
+
+const customStyles = {
```

Comment:
Instead of using inline CSS here we could use the componentType to set different classes, e.g.

```
className={classNames({
   'hue-overflow-item-breadcrumb': componentType === 'breadcrumb', 
   'hue-overflow-item-menu': componentType === 'menu'
 })}
```

BUT I question if an OverflowingItem component should need to know about this at all. We should let OverflowingItem handle the overflow related functionality only. So I suggest that we skip the whole `componentType` for this component. Instead you can allow the parent (PathBrowser etc) to set `className` and have then set the styling in the scss files, keeping this component clean from special styling.

What you also could do to take this to one step further is to create a hook for the actual overflow checking functionally so that we can use that hook in more components in the future.